qa/rpc-tests/p2p-leaktests.py

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env python3
+# Copyright (c) 2017 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+from test_framework.mininode import *
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import *
+import logging
+
+class CLazyNode(NodeConnCB):
+    def __init__(self):
+        NodeConnCB.__init__(self)
+        self.connection = None
+        self.done = False
+        self.unrequested_msg = False
+
+    def add_connection(self, conn):
+        self.connection = conn
+
+    def check(self):
+        def done():
+            return self.done
+        return wait_until(done, timeout=10) and not self.unrequested_msg
+
+    def send_message(self, message):
+        self.connection.send_message(message)
+
+    def bad_message(self, message):
+        self.unrequested_msg = True
+        raise Exception("should not have received message: %s" % message.command)
```

Comment:
I don't think you should be raising an exception in the network thread. Just set `unexpected_message` to True (and optionally print some debug info)